netty/src/main/java/io/grpc/netty/Utils.java

```diff
@@ -176,6 +184,78 @@ public static Status statusFromThrowable(Throwable t) {
     return s;
   }
 
+  /**
+   * Returns a {@link ChannelFactory} of default channel; {@code
+   * io.netty.channel.epoll.EpollSocketChannel} when Epoll is available, otherwise using {@link
+   * NioSocketChannel}.
+   */
+  @SuppressWarnings("unchecked")
+  static ChannelFactory<? extends Channel> defaultChannelFactory() {
```

Comment:
Instead of splitting the `isEpollAvailable()` logic between each of these methods, I'd much rather it was checked and processed once or twice (for client vs server). We also don't want to be doing the reflection every Channel creation.

It'd probably be best/easiest to put the logic in a `static` block.

Reply:
done.